### PR TITLE
When PTDS is enabled, print warning if the allocator is not ARENA

### PR DESCRIPTION
**Pull request by abellina**
Signed-off-by: Alessandro Bellina <abellina@nvidia.com>

Add a disclaimer for #1141 (`cudaErrorillegalAddress` when PTDS and DEFAULT are used) for now while we root cause it.

### CODE REVIEW

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala [hunk 1]
```diff
@@ -204,6 +204,11 @@ object GpuDeviceManager extends Logging {
       if (conf.isPooledMemEnabled) {
         init = conf.rmmPool match {
           case c if "default".equalsIgnoreCase(c) =>
+            if (Cuda.isPtdsEnabled) {
+              logWarning("Configuring the DEFAULT allocator with a CUDF built for " +
+                  "Per-thread Default Stream (PTDS). This is known to be unstable! " +
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala:209]:
```suggestion
                  "Per-Thread Default Stream (PTDS). This is known to be unstable! " +
```

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala [hunk 2]
```diff
@@ -352,7 +352,9 @@ object RapidsConf {
     .doc("Select the RMM pooling allocator to use. Valid values are \"DEFAULT\", \"ARENA\", and " +
       "\"NONE\". With \"DEFAULT\", `rmm::mr::pool_memory_resource` is used; with \"ARENA\", " +
       "`rmm::mr::arena_memory_resource` is used. If set to \"NONE\", pooling is disabled and RMM " +
-      "just passes through to CUDA memory allocation directly.")
+      "just passes through to CUDA memory allocation directly. Note: \"ARENA\" is the " +
+      "recommended pool allocator if CUDF is built with Per-thread Default Stream (PTDS), " +
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala:356]:
```suggestion
      "recommended pool allocator if CUDF is built with Per-Thread Default Stream (PTDS), " +
```

**Comment by sameerz** [comment]
build

**Comment by abellina** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**